gv.c

```diff
@@ -3376,6 +3376,46 @@ Perl_try_amagic_un(pTHX_ int method, int flags) {
     return FALSE;
 }
 
+/*
+=for apidoc amagic_find
+
+Check C<sv> for the overloaded (active magic) operation C<method>, and
+return the C<CV *> or C<NULL>.
+
+C<method> is one of the values found in F<overload.h>.
+
+C<flags> are available for future use.
+
+=cut
+*/
+CV *
+Perl_amagic_find(pTHX_ SV *sv, int method, int flags)
+{
+    PERL_ARGS_ASSERT_AMAGIC_FIND;
+    assert(flags == 0);
+    (void)flags;
+
+    if (!SvAMAGIC(sv))
+        return NULL;
+
+    HV *stash = SvSTASH(SvRV(sv));
+    if (!Gv_AMG(stash))
+        return NULL;
+
+    MAGIC *mg = mg_find((const SV *)stash, PERL_MAGIC_overload_table);
+    if (!mg)
+        return NULL;
+
+    CV **cvp = NULL;
+    if (AMT_AMAGIC((AMT *)mg->mg_ptr))
+        cvp = ((AMT *)mg->mg_ptr)->table;
+    if (!cvp)
+        return NULL;
+
+    CV *cv = cvp[method];
```

Comment:
There's no bounds check here, but perhaps at the top of the function it might be a good idea to add

```c
assert(method >= 0 && method < max_amg_code);
```

Reply:
Actually on further inspection there's a better macro for that.

```
assert(method >= 0 && method < NofAMmeth);
```